tests/core/full_node/stores/test_coin_store.py

```diff
@@ -161,22 +161,40 @@ async def test_set_spent(self, cache_size: uint32, db_version, bt):
                 if block.is_transaction_block():
                     removals: List[bytes32] = []
                     additions: List[Coin] = []
+                    async with db_wrapper.write_db():
+                        if block.is_transaction_block():
+                            assert block.foliage_transaction_block is not None
+                            await coin_store.new_block(
+                                block.height,
+                                block.foliage_transaction_block.timestamp,
+                                block.get_included_reward_coins(),
+                                additions,
+                                removals,
+                            )
 
-                    if block.is_transaction_block():
-                        assert block.foliage_transaction_block is not None
-                        await coin_store.new_block(
-                            block.height,
-                            block.foliage_transaction_block.timestamp,
-                            block.get_included_reward_coins(),
-                            additions,
-                            removals,
-                        )
-
-                    coins = block.get_included_reward_coins()
-                    records = [await coin_store.get_coin_record(coin.name()) for coin in coins]
+                        coins = block.get_included_reward_coins()
+                        records = [await coin_store.get_coin_record(coin.name()) for coin in coins]
 
                     await coin_store._set_spent([r.name for r in records], block.height)
 
+                    if len(records) > 0:
+                        for r in records:
+                            assert (await coin_store.get_coin_record(r.name)) is not None
+
+                        if cache_size > 0:
+                            # Check that we can't spend a coin twice in cache
+                            with pytest.raises(ValueError) as exec_info:
+                                await coin_store._set_spent([r2.name for r2 in records], block.height)
+                            assert "Coin already spent" in exec_info.value.args[0]
+
+                            for r3 in records:
+                                coin_store.coin_record_cache.remove(r3.name)
```

Comment:
is there a reason why you alternate between `r2` and `r3` for the iterator variable name? I was a bit confused at first, assuming those names were relevant. I think you should use the same name in all loops, and probably not have a number at all

Reply:
To avoid overlap in names, if you have:
```
for r in records:
    for r in records:
        ...
    print(r)
```
We would get the inner loop's r right?

Reply:
I see, so the number is the nesting level of `r`? in that case, all `r3` could be renamed `r2` I believe